sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/TracerProvider.java

```diff
@@ -289,10 +281,18 @@ public void addEvent(String name, Map<String, Object> attributes, OffsetDateTime
 
     private void end(int statusCode, Throwable throwable, Context context) {
         if (throwable != null) {
-            tracer.setAttribute(TracerProvider.ERROR_MSG, throwable.getMessage(), context);
-            tracer.setAttribute(TracerProvider.ERROR_TYPE, throwable.getClass().getName(), context);
+            if (statusCode == HttpConstants.StatusCodes.NOTFOUND) {
+                tracer.setAttribute(TracerProvider.ERROR_MSG, "Not found exception", context);
```

Comment:
I believe 404 statusCode is enough and the core tracer will populate all the info needed. 

https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/implementation/HttpTraceUtil.java#L179

Can we avoid populating custom error codes?

Reply:
If we remove ERROR_TYPE/ERROR_MSG then we wont be seeing below attributes which was the initial requirement from @trask
![image](https://user-images.githubusercontent.com/43345335/134070740-269413b1-9d3c-4ef4-937e-9ea25a4217b6.png)

Reply:
since this is no longer an exception, it shouldn't have exception attributes  - `otel.status_description` gives user the info they need

Reply:
> If we remove ERROR_TYPE/ERROR_MSG then we wont be seeing below attributes which was the initial requirement from @trask

this was probably before I understood the otel semantic convention for exceptions 😅

@lmolkova is correct that `recordException()` is the right way to attach the exception data